docs/blocks-storage/querier.md

```diff
@@ -109,6 +109,10 @@ querier:
   # CLI flag: -querier.ingester-metadata-streaming
   [ingester_metadata_streaming: <boolean> | default = true]
 
+  # Use LabelNames ingester RPCs with match params.
+  # CLI flag: -querier.ingester-label-names-with-matchers
+  [ingester_label_names_with_matchers: <boolean> | default = false]
```

Comment:
Do we need this config?
If the matcher is in the request, its not cause we always wanna filter out?

Reply:
This feature flag is for safe guarding during a deployment.
In this PR, queriers will start calling ingester LabelNames with Matchers. But if ingesters haven't restarted yet, then ingesters will ignore that and return label names without filtering out.


We can remove this flag in 2 releases and make this behavior the default.

Reply:
I suggest we add TODO or create an issue about removing/deprecating this flag now. Otherwise we will forget

Reply:
I was thinking that if the ingesters ignored the matchers in the request from the querier since the ingesters had not restarted yet, wouldn't that be the same behavior today since the ingesters already do not handle the matchers?

Reply:
Today's behavior for label names calls with matchers is like this:
- Querier will retrieve all the series using `MetricsForLabelMatchers` call. It'll then get the unique label names from the retrieved series.

The new behaviour is for label names calls with matchers:
- Querier will directly call `LabelNames` call and pass the matchers.
- If ingesters ignore the matchers, this will break the API.

I think having the feature flag is unavoidable.

Created https://github.com/cortexproject/cortex/issues/6222 for tracking the deprecation of the flag.

Reply:
I see, thank you for the clarification @harry671003.